### PR TITLE
feat: add puppet

### PR DESCRIPTION
**Pull request by amaanq**
cc @techsk8

### CODE REVIEW

File: queries/puppet/indents.scm [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  "}"
+  "]"
+  ")"
+] @indent.end
+
+[ "{" "}" ] @indent.branch
+
+[ "[" "]" ] @indent.branch
+
+[ "(" ")" ] @indent.branch
```
Comment (lucario387) [inline queries/puppet/indents.scm:25]: I'd suggest not setting opening brackets to `@indent.branch`, unless all of its parent's nodes are marked with `@indent.begin`

Comment (amaanq) [reply queries/puppet/indents.scm:25]: so just closed ones? or do you mean just for [ ]

Comment (lucario387) [reply queries/puppet/indents.scm:25]: Just closed ones. Opening ones are messier to handle correctly

File: queries/puppet/injections.scm [hunk 2]
```diff
@@ -0,0 +1,2 @@
+((regex) @regex
```
Comment (ObserverOfTime) [inline queries/puppet/injections.scm:1]: Missing comment injection.

File: queries/puppet/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+] @keyword.function
+
+[
+  "if"
+  "elsif"
+  "else"
+  "case"
+] @conditional
+
+(default_case "default" @conditional)
```
Comment (theHamsta) [inline queries/puppet/highlights.scm:35]: wouldn't default be `@label` in C/C++/Java/Dart?

**Review by theHamsta — APPROVED** [review]

File: queries/puppet/highlights.scm (hunk 3 above)
Comment (amaanq) [reply queries/puppet/highlights.scm:35]: Hmm, I don't particularly like that, it would fit better as being its own "case", i.e. a condition imo

**Comment by techsk8** [comment]
I love it so much, thank you @amaanq !!! You rock! 😍

**Comment by HerveMARTIN** [comment]
Thank you @amaanq !
Is there a way to test it before the next release? (I'm using LunarVim...)

**Comment by techsk8** [comment]
Yes, there's a way.
On what Platform are you using Lunarvim?

**Comment by HerveMARTIN** [comment]
I'm on arch linux.

**Comment by techsk8** [comment]
Well, then it's easy

```console
⚡➜ ~ cd .local/share/lunarvim/site/pack/lazy/opt/nvim-treesitter
⚡➜ nvim-treesitter ( master) git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
⚡➜ nvim-treesitter ( master) git pull
Already up to date.
⚡➜ nvim-treesitter ( master) lvim manifests/init.pp
```

**Comment by techsk8** [comment]
But I just installed the latest `lunarvim` version on a server and the puppet TS was automatically installed once I opened a `.pp` file, so you should be good if you're on the latest version
```
LV_BRANCH='release-1.3/neovim-0.9'
```

**Comment by amaanq** [comment]
off topic, take this issue to lunarvim

**Pull request merged**